admin/spec/service/check-complete.service.spec.js

```diff
@@ -1,61 +1,59 @@
-
-  afterEach(() => sandbox.restore())
-
-  function setupService (cb) {
-    const pupil = Object.assign({}, pupilMock)
-    spy = jasmine.createSpy('sqlAddResult').and.callFake(cb)
-    markingSpy = jasmine.createSpy('mark').and.callFake(cb)
-    sandbox.mock(pupilDataService).expects('findOne').resolves(pupil)
-    sandbox.mock(jwtService).expects('decode').resolves({ sub: '49g872ebf624b75400fbee09' })
-    spyOn(pupilDataService, 'update').and.returnValue(null)
-    return proxyquire('../../services/check-complete.service', {
-      './data-access/completed-check.data.service': {
-        sqlAddResult: spy
-      },
-      './marking.service': {
-        mark: markingSpy
-      },
-      '../../services/data-access/pupil.data.service': pupilDataService,
-      '../../services/jwt.service': jwtService
-    })
-  }
-
   describe('happy path', () => {
+    let service
+    const pupilMock = {
+      id: 1,
+      school: 9991001,
+      foreName: 'Pupil',
+      lastName: 'One',
+      pin: 4466,
+      dateOfBirth: moment('2000-12-31 00:00:00'),
+      urlSlug: 'EE882072-D3FC-46F6-84BC-691BFB1B5722',
+      checkOptions: {
+        speechSynthesis: false
+      },
+      isTestAccount: false
+    }
+    const completedCheck = {
+      data: {
+        pupil: {
+          checkCode: 'EE882072-D3FC-46F6-84BC-734BFB1B5733'
+        }
+      }
+    }
     beforeEach(() => {
-      service = setupService(function () { return Promise.resolve(null) })
+      service = require('../../services/check-complete.service')
+      spyOn(pupilDataService, 'sqlUpdate').and.returnValue(Promise.resolve())
+      spyOn(pupilDataService, 'sqlFindOneById').and.returnValue(Promise.resolve(pupilMock))
+      spyOn(completedCheckDataService, 'sqlAddResult').and.returnValue(Promise.resolve())
+      spyOn(markingService, 'mark').and.returnValue(Promise.resolve())
+      spyOn(jwtService, 'decode').and.returnValue({ sub: 1 })
     })
 
-    it('calls pupil data service update method when pin exists', async (done) => {
-      await service.completeCheck({ data: {} })
-      expect(pupilDataService.update).toHaveBeenCalledTimes(1)
+    it('clears pin and sets expiry when not test account', async (done) => {
+      await service.completeCheck(completedCheck)
+      expect(pupilDataService.sqlUpdate).toHaveBeenCalledTimes(1)
       done()
     })
 
-    it('calls the completed check data service', async (done) => {
-      await service.completeCheck({ data: {} })
-      expect(spy).toHaveBeenCalledTimes(1)
-      expect(markingSpy).toHaveBeenCalledTimes(1)
+    it('stores the check and marks it', async (done) => {
+      await service.completeCheck(completedCheck)
+      expect(completedCheckDataService.sqlAddResult).toHaveBeenCalledTimes(1)
+      expect(markingService.mark).toHaveBeenCalledTimes(1)
       done()
     })
 
-    it('adds a timestamp', async (done) => {
-      await service.completeCheck({ data: {} })
+    xit('adds a timestamp', async (done) => {
```

Comment:
Is this test commented out because of some other missing re-factor?

Reply:
this is commented out due to a pending conversation with you and George regarding best way to hook into the appended property.  Lets discuss tuesday, unless you or George want to refactor it monday and show me the approach